modules/build/src/test/scala/scala/build/tests/BuildOptionsTests.scala

```diff
@@ -28,6 +28,43 @@ class BuildOptionsTests extends munit.FunSuite {
     )
   }
 
+  test("-S 3.nightly option works") {
+    val options = BuildOptions(
+      scalaOptions = ScalaOptions(
+        scalaVersion = Some("3.nightly"),
+        scalaBinaryVersion = None,
+        supportedScalaVersionsUrl =
+          Some(
+            Random.alphanumeric.take(10).mkString("")
+          ) // invalid url, it should use defaults from Deps.sc
+      )
+    )
+    val scalaParams = options.scalaParams.orThrow
+    assert(
+      scalaParams.scalaVersion.startsWith("3") && scalaParams.scalaVersion.endsWith("-NIGHTLY"),
+      "-S 3.nightly argument does not lead to scala3 nightly build option"
+    )
+  }
+
+  test("-S 2.nightly option works") {
+    val options = BuildOptions(
+      scalaOptions = ScalaOptions(
+        scalaVersion = Some("2.nightly"),
+        scalaBinaryVersion = None,
+        supportedScalaVersionsUrl =
+          Some(
+            Random.alphanumeric.take(10).mkString("")
+          ) // invalid url, it should use defaults from Deps.sc
+      )
+    )
+    val scalaParams        = options.scalaParams.orThrow
+    val scala2NightlyRegex = raw"""(\d+)\.(\d+)\.(\d+)-bin-[a-f0-9]*""".r
```

Comment:
I think that regex for scala 2 nightly should only be defined once in code. I see, that was defined 3 times.

Reply:
What is the best place for defining it? I wanted to get it from coursier, but the api did not provide a means for that.

Reply:
@lwronski

Reply:
@alexarchambault do you think it is a good idea for now to create a pull request to add a function to Versions, as I have DM’ed you for solving this?

Reply:
Something like this:
```
def getBinaryType(scalaVersion: String): ScalaBinaryType =
    scalaVersion match {
      case Scala3EarlyVersion(milestone)         =>  ScalaBinaryType.Scala3EarlyVersion
      case Scala3Version(_, _)                   =>  ScalaBinaryType.Scala3
      case ReleaseVersion(major, minor, _)       => ScalaBinaryType.ReleaseVersion
      case MinorSnapshotVersion(major, minor, _) => ScalaBinaryType.MinorSnapshotVersion
      case NightlyVersion(major, minor, _)       => ScalaBinaryType.NightlyVersion
      case DottyVersion(minor, _)                => ScalaBinaryType.DottyVersion
      case TypelevelVersion(major, minor, _)     => ScalaBinaryType.TypeLevelVersion
      case _                                     => scalaVersion
    }
```

Reply:
I would be doing that. Just a moment.

Reply:
I have put it in the BuildOptions companion object, since it is only related to BuildOptions.